numpy/__init__.pyi

```diff
@@ -540,38 +572,64 @@ def vsplit(ary, indices_or_sections): ...
 def who(vardict=...): ...
 
 # Placeholders for C-based functions
-arange: Callable[..., Any]
-bincount: Callable[..., Any]
-busday_count: Callable[..., Any]
-busday_offset: Callable[..., Any]
-can_cast: Callable[..., Any]
-compare_chararrays: Callable[..., Any]
-concatenate: Callable[..., Any]
-copyto: Callable[..., Any]
-datetime_as_string: Callable[..., Any]
-datetime_data: Callable[..., Any]
-dot: Callable[..., Any]
-frombuffer: Callable[..., Any]
-fromfile: Callable[..., Any]
-fromiter: Callable[..., Any]
-frompyfunc: Callable[..., Any]
-fromstring: Callable[..., Any]
-geterrobj: Callable[..., Any]
-inner: Callable[..., Any]
-is_busday: Callable[..., Any]
-lexsort: Callable[..., Any]
-may_share_memory: Callable[..., Any]
-min_scalar_type: Callable[..., Any]
-nested_iters: Callable[..., Any]
-packbits: Callable[..., Any]
-promote_types: Callable[..., Any]
-putmask: Callable[..., Any]
-result_type: Callable[..., Any]
-seterrobj: Callable[..., Any]
-shares_memory: Callable[..., Any]
-unpackbits: Callable[..., Any]
-vdot: Callable[..., Any]
-where: Callable[..., Any]
+# TODO: Sort out which parameters are positional-only
+@overload
+def arange(stop, dtype=..., *, like=...): ...
+@overload
+def arange(start, stop, step, dtype=..., *, like=...): ...
+def bincount(x, weights=..., minlength=...): ...
+def busday_count(
+    begindates,
+    enddates,
+    weekmask=...,
+    holidays=...,
+    busdaycal=...,
+    out=...,
+): ...
+def busday_offset(
+    dates,
+    offsets,
+    roll=...,
+    weekmask=...,
+    holidays=...,
+    busdaycal=...,
+    out=...,
+): ...
+def can_cast(from_, to, casting=...): ...
+def compare_chararrays(a, b, cmp_op, rstrip): ...
+def concatenate(__a, axis=..., out=..., dtype=..., casting=...): ...
+def copyto(dst, src, casting=..., where=...): ...
+def datetime_as_string(arr, unit=..., timezone=..., casting=...): ...
+def datetime_data(__dtype): ...
```

Comment:
The double leading underscore is used for denoting position-only arguments per [PEP 484](https://www.python.org/dev/peps/pep-0484/#positional-only-arguments). The newer `/`-based syntax introduced in [PEP 570](https://www.python.org/dev/peps/pep-0570/) is, unfortunatelly, not yet supported by mypy for python versions prior to python 3.8.